File: sum of sub array/sub.py
#Given an array arr[] of non-negative integers and an integer sum, find a subarray that adds to a given sum.

#Note: There may be more than one subarray with sum as the given sum, print first such subarray. 
def sub(arr,sum):
    curr_sum = arr[0]
    start = 0
    n = len(arr)
    for i in range(1,n):
        while curr_sum > sum and start < i-1:
            curr_sum -= arr[start]
            start += 1
        if curr_sum == sum:
            return arr[start:i]
        curr_sum += arr[i]
    return -1
print(sub([1,2,3,7,5],12))

File: sum of sub array/sub.js
//Given an array arr[] of non-negative integers and an integer sum, find a subarray that adds to a given sum.

//Note: There may be more than one subarray with sum as the given sum, print first such subarray. 

function sub(arr,sum){

    let curr_sum=arr[0]
    let start=0
    for(let i=1;i<arr.length;i++){
        while(curr_sum>sum && start<i-1){
            curr_sum=curr_sum-arr[start]
            start++
        }
        if(curr_sum==sum){
            return arr.slice(start,i)
        }
        if(i<arr.length){
            curr_sum=curr_sum+arr[i]
        }
    }
    return -1
}
console.log(sub([1,2,3,7,5],12))